src/chaosinventory/authentication/models.py

```diff
@@ -0,0 +1,37 @@
+import secrets
+import uuid
+
+from django.contrib.auth.models import User
+from django.db import models
+
+
+class Token(models.Model):
+    id = models.UUIDField(
+        primary_key=True,
+        default=uuid.uuid4,
+        editable=False
+    )
+    key = models.CharField(max_length=40)
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE
+    )
+    created = models.DateTimeField(auto_now_add=True)
+    application = models.CharField(max_length=255)
+    expiring = models.DateTimeField(
+        null=True,
+        blank=True
+    )
```

Comment:
Currently, it is possible to set expiring dates in the past. Whilst this mostly is an issue for the apps providing a wrong date, we could in the future catch this in the validation process.

Reply:
I'd suggest opening an issue for later validation improvement, as it is no blocking or security relevant issue for now?